frontend/src/Types.elm

```diff
@@ -0,0 +1,196 @@
+module Types exposing (..)
+
+{-| The types used in the application, importantly
+
+* Model, our model of the world
+* Msg, all the actions the app can perform
+
+These are in a separate file because they need to be imported in multiple files. Any other place
+for the types would lead to circular dependencies.
+-}
+
+import Bootstrap.Navbar as Navbar
+import Navigation
+import Json.Decode as Decode exposing (string, bool, int, float)
+import Json.Decode.Pipeline as Decode exposing (..)
+import Json.Encode as Encode
+import Http
+
+
+{-| Our model of the world
+-}
+type alias Model =
+    { route : Route
+    , navbarState : Navbar.State
+    , profiling : ProfilingState
+    , attribution : AttributionState
+    }
+
+
+{-| All the actions our application can perform
+-}
+type Msg
+    = NoOp
+    | NavbarMsg Navbar.State
+    | ChangeRoute Route
+    | AttributionMsg AttributionMessage
+    | ProfilingMsg ProfilingMessage
+    | UrlChange Navigation.Location
+
```

Comment:
You are explaining what all this stuff means in another file (Update), but it wouldn't hurt to also explain it here. While for someone familiar with the project these are quite self-explanatory, someone seeing this for the first time might not feel the same way.